File: hsg9-tinh/2022-2023-thuathienhue/ngaycong/ngaycong.py
import os
import sys

input_file = os.path.join(sys.path[0], 'ngaycong.inp')
output_file = os.path.join(sys.path[0], 'ngaycong.out')

n = 0
days = []

total = 0
average = 0.0


def input():
    global n, days

    with open(input_file, 'r') as f:
        n = int(f.readline())

        days = list(map(int, f.readline().split()))


def process():
    global n, days, total, average

    # Sắp xếp ngày công giảm dần
    days.sort(reverse=True)
    
    # Tính tổng số ngày
    total = sum(days)

    # Tính trung bình cộng
    average = total / n


def output():
    global days, total, average

    with open(output_file, 'w') as fo:
        fo.write(' '.join(map(str, days)) + '\n')

        fo.write(f'{total} {average:.2f}')


if __name__ == '__main__':
    input()
    process()
    output()

File: hsg9-tinh/2022-2023-thuathienhue/ngaycong/ngaycong.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
#include <iomanip>

#define input_file "ngaycong.inp"
#define output_file "ngaycong.out"

using namespace std;

int n;
vector<int> days;

int total;
float average;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> n;

    days.resize(n, 0);

    for (int i = 0; i < n; ++i)
    {
        cin >> days[i];
    }
}


void process()
{
    // Sắp xếp ngày công giảm dần
    sort(days.begin(), days.end(), greater<int>());

    // Tính tổng số ngày
    total = accumulate(days.begin(), days.end(), 0);

    // Tính trung bình cộng
    average = (float) total / days.size();
}


void output()
{
    freopen(output_file, "w", stdout);

    for (int i = 0; i < n; ++i)
    {
        cout << days[i] << ' ';
    }

    cout << '\n';

    cout << total << ' ' << fixed << setprecision(2) << average;
}


int main()
{
    input();
    process();
    output();

    return 0;
}